src/file-manager/fm-desktop-icon-view.c

```diff
@@ -386,6 +393,97 @@ fm_desktop_icon_view_handle_middle_click (CajaIconContainer *icon_container,
                 ButtonPressMask, (XEvent *) &x_event);
 }
 
+#ifdef HAVE_WAYLAND
+static void
+update_background_on_wayland(FMDesktopIconView *desktop_icon_view)
+{
+    /*Draw and update the background using GTK when in wayland*/
+    char            *tmp;
+    gchar           *filename;
+    gchar           *css;
+    GString         *string;
+    GtkWidget       *toplevel;
+    GtkStyleContext *context;
+    GtkCssProvider *provider;
+
+    /* We call this from here due to "Change Desktop Background"
+     * being part of this file but we draw the background
+     * on the TOPLEVEL window not the icon view within it
+     */
+
+    g_return_if_fail (GDK_IS_WAYLAND_DISPLAY (gdk_display_get_default ()));
+
+    toplevel = gtk_widget_get_toplevel(GTK_WIDGET(desktop_icon_view));
+    filename = NULL;
+    tmp = g_settings_get_string (background_settings, "picture-filename");
```

Comment:
Memory leak

Reply:
`tmp ` removed entirely in latest force-push. This indirect route was a leftover from my old 2016 work that used GTK to provide a fallback background directly in the non-composited case (similar to the no root window situation with wayland) when we had experiemented with a nemo/nautilus style transparent desktop to fix an ugly redraw problem. When that was traced to deprecated code that had broken, those changes were reverted but as they were still in  the git commit records they gave me a template for wayland background management.